File: maven-samples/javafx/src/main/java/equochromium/SampleJavaFx.java
package equochromium;

import com.equo.chromium.ChromiumBrowser;

import javafx.application.Application;
import javafx.geometry.Insets;
import javafx.geometry.Pos;
import javafx.scene.Scene;
import javafx.scene.control.Button;
import javafx.scene.control.Label;
import javafx.scene.control.TextField;
import javafx.scene.layout.BorderPane;
import javafx.scene.layout.HBox;
import javafx.scene.layout.Priority;
import javafx.scene.layout.StackPane;
import javafx.stage.Stage;

/**
 * JavaFX browser sample — embeds an Equo Chromium browser (offscreen-rendered) as a real
 * {@code javafx.scene.Node} inside a JavaFX scene graph, alongside native JavaFX controls.
 *
 * <pre>mvn verify</pre>
 * launches it (override the page with {@code -Durl=...}). The browser lives in the CENTER of a
 * {@link BorderPane}, under a JavaFX toolbar — proving it respects JavaFX layout/clipping.
 */
public class SampleJavaFx extends Application {

	@Override
	public void start(Stage stage) {
		String url = System.getProperty("url", "https://docs.equo.dev/main/getting-started/introduction.html");

		// The browser is embedded as a JavaFX Node inside this host pane.
		StackPane browserHost = new StackPane();
		ChromiumBrowser browser = ChromiumBrowser.javafx(browserHost, url);

		// A real JavaFX toolbar around the browser, to show scene-graph integration.
		Label brand = new Label("Equo Chromium — JavaFX");
		TextField address = new TextField(url);
		HBox.setHgrow(address, Priority.ALWAYS);
		address.setOnAction(e -> browser.setUrl(address.getText()));

		Button devBtn = new Button("DevTools");
		devBtn.setOnAction(e -> browser.showDevTools());

		HBox toolbar = new HBox(10, brand, address, devBtn);
		toolbar.setAlignment(Pos.CENTER_LEFT);
		toolbar.setPadding(new Insets(8));

		BorderPane root = new BorderPane();
		root.setTop(toolbar);
		root.setCenter(browserHost);

		stage.setScene(new Scene(root, 1100, 760));
		stage.setTitle("Equo Chromium — JavaFX sample");
		stage.show();
	}

	public static void main(String[] args) {
		launch(args);
	}
}

File: gradle-samples/javafx/app/src/main/kotlin/SampleJavaFx/SampleJavaFx.kt
package SampleJavaFx

import com.equo.chromium.ChromiumBrowser
import javafx.application.Application
import javafx.geometry.Insets
import javafx.geometry.Pos
import javafx.scene.Scene
import javafx.scene.control.Label
import javafx.scene.control.TextField
import javafx.scene.layout.BorderPane
import javafx.scene.layout.HBox
import javafx.scene.layout.Priority
import javafx.scene.layout.StackPane
import javafx.stage.Stage

/**
 * JavaFX browser sample — embeds an Equo Chromium browser (offscreen-rendered) as a real
 * javafx.scene.Node inside a JavaFX scene graph, next to native JavaFX controls.
 *
 * Run: ./gradlew :app:run  (override the page with -Durl=...)
 */
class SampleJavaFx : Application() {
    override fun start(stage: Stage) {
        val url = System.getProperty("url", "https://docs.equo.dev/main/getting-started/introduction.html")

        // The browser is embedded as a JavaFX Node inside this host pane.
        val browserHost = StackPane()
        val browser = ChromiumBrowser.javafx(browserHost, url)

        // A real JavaFX toolbar around the browser, to show scene-graph integration.
        val address = TextField(url)
        HBox.setHgrow(address, Priority.ALWAYS)
        address.setOnAction { browser.setUrl(address.text) }
        val toolbar = HBox(10.0, Label("Equo Chromium — JavaFX"), address)
        toolbar.alignment = Pos.CENTER_LEFT
        toolbar.padding = Insets(8.0)

        val root = BorderPane()
        root.top = toolbar
        root.center = browserHost

        stage.scene = Scene(root, 1100.0, 760.0)
        stage.title = "Equo Chromium — JavaFX sample"
        stage.show()
    }
}

// Non-Application entry point: compiled to class SampleJavaFx.SampleJavaFxKt so the JVM main
// class is NOT an Application subclass (avoids the "JavaFX runtime components are missing" error).
fun main() {
    Application.launch(SampleJavaFx::class.java)
}
